kvserver: unskip v1 flow control integration tests under duress

kvoli opened the pull request:
Only the last commit is relevant.

---

These were skipped to unblock merging https://github.com/cockroachdb/cockroach/pull/132125 and later (presumed to
be) fixed by https://github.com/cockroachdb/cockroach/pull/132563.

Un-skip all `TestFlowControl.*` v1 integration tests under duress.

Resolves: https://github.com/cockroachdb/cockroach/issues/132310
Release note: None

kvoli committed:
kvserver: unskip v1 flow control integration tests under duress

These were skipped to unblock merging #132125 and later (presumed to
be) fixed by #132563.

Un-skip all `TestFlowControl.*` v1 integration tests under duress.

Resolves: #132310
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132570)
<!-- Reviewable:end -->

kvoli committed:
testutils,kvserver: disable store rebalancer when replication manual

`ReplicationManual` can be specified in tests which use a test cluster
to prevent lease/replica movements.

This proves vital if the test is asserting exactly on the placement of a
range, or its leaseholder.

Also disable the store rebalancer, which could move both leases and
replicas under `ReplicationManual` before.

Informs: #132310
Informs: #132272
Release note: None

sumeerbhola commented:
Looks like there is a failure in `TestFlowControlAdmissionPostSplitMerge` under stressrace.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132570#-:-O9BYv0q3f7y_PwLbp6L:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @pav-kv)



<!-- Sent from Reviewable.io -->

kvoli commented:
Hmmm, CI is now failing again, where previously it was passing on these commits.

I'm going to wait until the first commit merges, rebase and then check if this is still true.

kvoli commented:
TYFTR!

bors r=sumeerbhola
